retrofit/src/main/java/retrofit2/ServiceMethod.java

```diff
@@ -51,62 +52,81 @@
 import retrofit2.http.QueryMap;
 import retrofit2.http.Url;
 
-import static retrofit2.Utils.methodError;
+/** Adapts an invocation of an interface method into an HTTP call. */
+final class ServiceMethod<T> {
+  // Upper and lower characters, digits, underscores, and hyphens, starting with a character.
+  private static final String PARAM = "[a-zA-Z][a-zA-Z0-9_-]*";
+  private static final Pattern PARAM_URL_REGEX = Pattern.compile("\\{(" + PARAM + ")\\}");
+  private static final Pattern PARAM_NAME_REGEX = Pattern.compile(PARAM);
+
+  public final okhttp3.Call.Factory callFactory;
+  public final CallAdapter<?> callAdapter;
 
-final class RequestFactory {
-  private final String method;
   private final BaseUrl baseUrl;
+  private final Converter<ResponseBody, T> responseConverter;
+  private final String httpMethod;
   private final String relativeUrl;
   private final Headers headers;
   private final MediaType contentType;
   private final boolean hasBody;
   private final boolean isFormEncoded;
   private final boolean isMultipart;
-  private final ParameterAction[] parameterActions;
-
-  RequestFactory(String method, BaseUrl baseUrl, String relativeUrl, Headers headers,
-      MediaType contentType, boolean hasBody, boolean isFormEncoded, boolean isMultipart,
-      ParameterAction[] parameterActions) {
-    this.method = method;
-    this.baseUrl = baseUrl;
-    this.relativeUrl = relativeUrl;
-    this.headers = headers;
-    this.contentType = contentType;
-    this.hasBody = hasBody;
-    this.isFormEncoded = isFormEncoded;
-    this.isMultipart = isMultipart;
-    this.parameterActions = parameterActions;
+  private final ParameterHandler<?>[] parameterHandlers;
+
+  ServiceMethod(Builder<T> builder) {
+    this.callFactory = builder.retrofit.callFactory();
+    this.callAdapter = builder.callAdapter;
+    this.baseUrl = builder.retrofit.baseUrl();
+    this.responseConverter = builder.responseConverter;
+    this.httpMethod = builder.httpMethod;
+    this.relativeUrl = builder.relativeUrl;
+    this.headers = builder.headers;
+    this.contentType = builder.contentType;
+    this.hasBody = builder.hasBody;
+    this.isFormEncoded = builder.isFormEncoded;
+    this.isMultipart = builder.isMultipart;
+    this.parameterHandlers = builder.parameterHandlers;
```

Comment:
All of these fields need to be made package scope. Otherwise you're adding a synthetic method for each.